backend/onyx/connectors/connector_runner.py

```
@@ -25,6 +25,24 @@
 CT = TypeVar("CT", bound=ConnectorCheckpoint)
 
 
+def batched_docs(
+    checkpoint_connector_generator: CheckpointOutput[CT],
+    batch_size: int,
+) -> Generator[list[Document], None, None]:
+    batch: list[Document] = []
```

nit: can we add a comment letting people know we are discarding failures